PyInstaller/building/api.py

```diff
@@ -381,7 +382,14 @@ def __init__(self, *args, **kwargs):
             if isinstance(arg, TOC):
                 self.toc.extend(arg)
             elif isinstance(arg, Target):
-                self.toc.append((os.path.basename(arg.name), arg.name, arg.typ))
+                if hasattr(arg, '_EXE'):
```

Comment:
`Targets` may by PYZ, PKG, EXE and COLLECT. Can all these have an attribute `_EXE`?